doc/tutorials/acrn-secure-boot-with-grub.rst

```diff
@@ -0,0 +1,256 @@
+
+From a secureboot point of view:
+
+- UEFI firmware verifies shim/GRUB
+- GRUB verifies ACRN, Service VM kernel, and pre-launched User VM kernel
+- Service VM OS kernel verifies the Device Model (``acrn-dm``) and User
+  VM OVMF bootloader (with the help of ``acrn-dm``)
+- User VM virtual bootloader (e.g. OVMF) starts the guest side verified boot process
+
+This document shows you how to enable GRUB to
+verify ACRN binaries such ``acrn.bin``, Service VM kernel (``bzImage``), and
+if present, a pre-launched User VM kernel image.
+
+.. rst-class:: numbered-step
+
+Generate GPG Key
+****************
+
+GRUB supports loading GPG signed files only if digital signatures are
+enabled. Here's an example of generating a GPG signing key::
+
+  mkdir --mode 0700 keys
+  gpg --homedir keys  --gen-key
+  gpg --homedir keys --export > boot.key
+
+The :command:`gpg --gen-key` generates a public and private key pair.
+The private key is used to sign GRUB configuration files and ACRN
+binaries. The public key will be embedded in GRUB and is used to verify
+GRUB configuration files or binaries GRUB tries to load.
+
+.. rst-class:: numbered-step
+
+Setup Standalone GRUB EFI Binary
+********************************
+
+Prepare Initial GRUB Configuration grub.init.cfg
+================================================
+
+Create file ``grub.init.cfg`` to store the following minimal GRUB
+configuration.  The environment variable ``check_signatures=enforce``
+tells GRUB to enable digital signatures::
+
+    set check_signatures=enforce
+    export check_signatures
+
+    search --no-floppy --fs-uuid --set=root ESP_UUID
+    configfile /grub.cfg
+    echo /grub.cfg did not boot the system, rebooting in 10 seconds.
+    sleep 10
+    reboot
+
+Replace the ESP_UUID with the UUID of your EFI system partition (found
+by running the :command:`lsblk -f`.  In the example output below,
+the UUID is ``24FC-BE7A``:
+
+.. code-block:: console
+    :emphasize-lines: 2
+
+    sda
+    ├─sda1 vfat   ESP    24FC-BE7A                            /boot/efi
+    ├─sda2 vfat   OS     7015-557F
+    ├─sda3 ext4   UBUNTU e8640994-b2a3-45ad-9b72-e68960fb22f0 /
+    └─sda4 swap          262d1113-64be-4910-a700-670b9d2277cc [SWAP]
+
+
+Enable Authentication in GRUB
+=============================
+
+With authentication enabled, a user/password is required to restrict
+access to the GRUB shell, where arbitrary commands could be run.
+A typical GRUB configuration fragment (added to ``grub.init.cfg``) might
+look like this::
+
+    set superusers="root"
+    export superusers
+    password_pbkdf2 root GRUB_PASSWORD_HASH
+
+Replace the ``GRUB_PASSWORD_HASH`` with the result of the :command:`grub-mkpasswd-pbkdf2`
+with your custom passphrase.
+
+Use this command to sign the :file:`grub.init.cfg` file with your private
+GPG key and create the :file:`grub.init.cfg.sig`::
+
+    gpg --homedir keys --detach-sign grub.init.cfg
+
+
+Create a single GRUB EFI Binary
```

Comment:
I'm thinking this should be ``signed`` and not ``single`` here, and on line 126 below

Reply:
@dbkinder, not signed, suppose "standalone" is better, right?